Alteration to salary

raecyst opened the pull request:
Allows classes to have salaries, as per suggestion by Misfit. Feel this can be useful for different ranks/divisions being paid differently.

raecyst committed:
Alteration to salary

Allows classes to have salaries, as per suggestion by Misfit. Feel this can be useful for different ranks/divisions being paid differently.

TovarischPootis commented on gamemode/core/hooks/sv_hooks.lua:
```diff
@@ -722,7 +722,8 @@ function GM:CreateSalaryTimer(client)
 	local faction = nut.faction.indices[character:getFaction()]
-	if (not faction or not isnumber(faction.pay) or faction.pay <= 0) then
+	local class = nut.class.indices[character:getClass()]
+	if ((not faction or not isnumber(faction.pay) or faction.pay <= 0) and (not class or not isnumber(class.pay) or class.pay <= 0)) then
```
Wouldn't this return if the character is not part of a class or the class in question doesn't have a custom pay?
By default, characters are not assigned to a class. That will pass in the "not class" and simply return.
If the character is part of a class, but its pay is not defined (for instance the class is supposed to use the default faction pay), it will pass the "not isnumber(class.pay) or class.pay <= 0)" and also return

raecyst committed:
Fixed class commit

Fixed class salary commit

raecyst committed:
Another quick fix

Simplifies the return if there's no money to give.

TovarischPootis commented on gamemode/core/hooks/sv_hooks.lua:
```diff
@@ -721,10 +721,12 @@ function GM:CreateSalaryTimer(client)
+	local faction = nut.faction.indices[character:getFaction()] or {}
+	local class = nut.class.indices[character:getClass()] or {}
+	
+	local pay = (class and class.pay) or (faction and faction.pay) or nil --thank you Tov.
```
```suggestion
	local pay = hook.Run("GetSalaryAmount", client, faction, class) or (class and class.pay) or (faction and faction.pay) or nil --thank you Tov.
```

raecyst committed:
Update gamemode/core/hooks/sv_hooks.lua

Works better in the long run.

Co-authored-by: TovarischPootis <54110479+TovarischPootis@users.noreply.github.com>

TovarischPootis commented on gamemode/core/hooks/sv_hooks.lua:
```diff
@@ -721,10 +721,12 @@ function GM:CreateSalaryTimer(client)
-	if (not faction or not isnumber(faction.pay) or faction.pay <= 0) then
-		return
-	end
+	local faction = nut.faction.indices[character:getFaction()] or {}
```
```suggestion
	local faction = nut.faction.indices[character:getFaction()] or nil
```

TovarischPootis commented on gamemode/core/hooks/sv_hooks.lua:
```diff
@@ -721,10 +721,12 @@ function GM:CreateSalaryTimer(client)
-		return
-	end
+	local faction = nut.faction.indices[character:getFaction()] or {}
+	local class = nut.class.indices[character:getClass()] or {}
```
```suggestion
	local class = nut.class.indices[character:getClass()] or nil
```

TovarischPootis reviewed (changes requested):
An empty table would still return true in an if check, which may cause errors

raecyst committed:
Update gamemode/core/hooks/sv_hooks.lua

Co-authored-by: TovarischPootis <54110479+TovarischPootis@users.noreply.github.com>

raecyst committed:
Update gamemode/core/hooks/sv_hooks.lua

Co-authored-by: TovarischPootis <54110479+TovarischPootis@users.noreply.github.com>

raecyst committed:
Hook to hide players in Scoreboard

A very simple hook that hides certain factions in the scoreboard if conditions are met. Good for hiding admin factions or event factions.

raecyst committed:
Merge branch '1.2-wip' of https://github.com/SeamusMcGowan/NutScript-1 into 1.2-wip

raecyst committed:
Optimization

Figured out how to optimize the hiding code.

raecyst committed:
Update cl_scoreboard.lua

Another small fix to the code, prevents faction appearing if amount is 0

TovarischPootis reviewed (commented):
This pull became 2 things
1st: Made class specific pay possible, all issues fixed
2nd: Adds a hook that allows factions to not be visible on the scoreboard, looks simple enough

zoephix committed:
Update sv_hooks.lua

zoephix committed:
Update cl_scoreboard.lua

TovarischPootis commented on plugins/scoreboard/derma/cl_scoreboard.lua:
```diff
@@ -98,7 +98,7 @@ local PANEL = {}
 				
 				if (amount != 0) then
-					v:SetVisible(hook.Run("HideFactionInScoreboard",faction) != false or LocalPlayer():IsAdmin())
+					v:SetVisible(hook.Run("HideFactionInScoreboard", faction) or LocalPlayer():IsAdmin())
```
`hook.Run("HideFactionInScoreboard", faction)` by itself would return nil, which is not what we want as that would cause nothing to draw on the scoreboard unless the player is an admin, we want to specifically check if false was returned or not.

TovarischPootis commented on plugins/scoreboard/derma/cl_scoreboard.lua:
```diff
@@ -94,6 +94,12 @@ local PANEL = {}
+				local faction = nut.faction.indices[k]
+				
+				if (amount != 0) then
+					v:SetVisible(hook.Run("HideFactionInScoreboard", faction) or LocalPlayer():IsAdmin())
```
```suggestion
					v:SetVisible(hook.Run("HideFactionInScoreboard", faction) ~= false or LocalPlayer():IsAdmin())
```

zoephix replied:
why

TovarischPootis replied:
https://github.com/NutScript/NutScript/pull/42#discussion_r605895303
that's why

raecyst replied:
Should probably change HideFactionInScoreboard to ShowFactionInScoreboard, since you return false to hide it.

raecyst committed:
Update plugins/scoreboard/derma/cl_scoreboard.lua

Co-authored-by: TovarischPootis <54110479+TovarischPootis@users.noreply.github.com>

TovarischPootis commented on plugins/scoreboard/derma/cl_scoreboard.lua:
```diff
@@ -94,6 +94,12 @@ local PANEL = {}
+				local faction = nut.faction.indices[k]
+				
+				if (amount != 0) then
+					v:SetVisible(hook.Run("HideFactionInScoreboard", faction) ~= false or LocalPlayer():IsAdmin())
```
```suggestion
					v:SetVisible(hook.Run("ShowFactionInScoreboard", faction) ~= false or LocalPlayer():IsAdmin())
```
To make the purpose and syntax of the hook clearer

TovarischPootis replied:
As per Seamus' suggestion

raecyst committed:
Update plugins/scoreboard/derma/cl_scoreboard.lua

Co-authored-by: TovarischPootis <54110479+TovarischPootis@users.noreply.github.com>
